### PR TITLE
[Solana] support block parsing

### LINKED ISSUE #1361: [Solana] block parsing
RPC:

1. `getConfirmedBlock`
2. `getEpochInfo`

### PR DESCRIPTION
**Pull request by hewigovens**
Fixes #1361 

Changes:
- Add block parsing
- Code cleanup, staking api will be implemented in client

### CODE REVIEW

File: platform/solana/transaction.go [hunk 1]
```diff
@@ -2,43 +2,66 @@ package solana
...
 	results := make(types.TxPage, 0)
-	txs, err := p.client.GetTransactions(address)
+	txs, err := p.client.GetTransactionsByAddress(address)
 	if err != nil {
 		return results, err
 	}
 	for _, tx := range txs {
-		if normalized, err := p.NormalizeTx(tx, address); err == nil {
+		timestamp := tx.BlockTime
+		if timestamp == 0 {
```
Comment (vikmeup) [inline platform/solana/transaction.go:29]: in which cases block time is 0?

File: platform/solana/transaction.go [hunk 2]
```diff
@@ -2,43 +2,66 @@ package solana
...
+			timestamp = EstimateTimestamp(tx.Slot)
```
Comment (vikmeup) [inline platform/solana/transaction.go:30]: can we remove `EstimateTimestamp` ? it's not accurate

File: platform/solana/block.go [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	return p.client.GetLasteBlock()
+}
+
+func (p *Platform) GetBlockByNumber(num int64) (*types.Block, error) {
+	block, err := p.client.GetTransactionsInBlock(num)
+	if err != nil {
+		return nil, err
+	}
+
+	page := make(types.TxPage, 0)
```
Comment (vikmeup) [inline platform/solana/block.go:15]: `types.TxPage` can we rename this to `types.Txs` ?

File: platform/solana/transaction.go (hunk 1 above)
Comment (hewigovens) [reply platform/solana/transaction.go:29]: this `field` is newly added, old data might not have it

Comment (vikmeup) [reply platform/solana/transaction.go:29]: let's skip such transactions instead

File: platform/solana/transaction.go (hunk 2 above)
Comment (hewigovens) [reply platform/solana/transaction.go]: done

File: platform/solana/transaction.go [hunk 4]
```diff
@@ -2,43 +2,65 @@ package solana
...
 	if err != nil {
 		return results, err
 	}
 	for _, tx := range txs {
-		if normalized, err := p.NormalizeTx(tx, address); err == nil {
+		if tx.BlockTime == 0 {
+			continue
+		}
+		if normalized, err := p.NormalizeTx(tx, tx.Slot, tx.BlockTime); err == nil {
+			if address == normalized.From {
```
Comment (vikmeup) [inline platform/solana/transaction.go:32]: can we you add extension method to transaction `GetDirection() -> Direction` ?  to avoid having this logic here

**Review by vikmeup — APPROVED** [review]
Couple of suggestions, otherwise lgtm

File: platform/solana/transaction.go (hunk 4 above)
Comment (hewigovens) [reply platform/solana/transaction.go]: thanks, there is an existing method to get direction

**Pull request merged**